### PR TITLE
Fix loss of new line chars when converting msg with GZIP HTTP responses into an ApiResponseSet

### PR DESCRIPTION
**Pull request by DiogoMRSilva**
Closes https://github.com/zaproxy/zaproxy/issues/7171

I couldn't find any changelog file, let me know if the PR misses something.

### CODE REVIEW

File: zap/src/main/java/org/zaproxy/zap/extension/api/ApiResponseConversionUtils.java [hunk 1]
```diff
@@ -98,12 +98,8 @@ private ApiResponseConversionUtils() {}
...
                     InputStreamReader isr = new InputStreamReader(gis);
                     BufferedReader br = new BufferedReader(isr); ) {
-                StringBuilder sb = new StringBuilder();
-                String line = null;
-                while ((line = br.readLine()) != null) {
-                    sb.append(line);
-                }
-                map.put("responseBody", sb.toString());
+                String response = org.apache.commons.io.IOUtils.toString(br);
+                map.put("responseBody", response);
```
[the remark this answers is not in the document]
Comment (DiogoMRSilva) [reply zap/src/main/java/org/zaproxy/zap/extension/api/ApiResponseConversionUtils.java:102]: Can you explain me why? All the tests pass, I need to understand to change the ones that need the charset updated.

[the remark this answers is not in the document]
Comment (DiogoMRSilva) [reply zap/src/main/java/org/zaproxy/zap/extension/api/ApiResponseConversionUtils.java]: Thx for the explanation :)

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by kingthorin — APPROVED** [review]
Thanks!

**Pull request merged**